openc3/lib/openc3/script/commands.rb

```diff
@@ -120,17 +120,25 @@ def _cmd(cmd, cmd_no_hazardous, *args, timeout: nil, log_message: nil, validate:
         _cmd_disconnect(cmd, raw, no_range, no_hazardous, *args, scope: scope)
       else
         begin
-          target_name, cmd_name, cmd_params = $api_server.method_missing(cmd, *args, timeout: timeout, log_message: log_message, validate: validate, scope: scope, token: token)
-          if log_message.nil? or log_message
-            _log_cmd(target_name, cmd_name, cmd_params, raw, no_range, no_hazardous)
+          begin
+            target_name, cmd_name, cmd_params = $api_server.method_missing(cmd, *args, timeout: timeout, log_message: log_message, validate: validate, scope: scope, token: token)
+            if log_message.nil? or log_message
+              _log_cmd(target_name, cmd_name, cmd_params, raw, no_range, no_hazardous)
+            end
+          rescue HazardousError => e
+            # This opens a prompt at which point they can cancel and stop the script
+            # or say Yes and send the command. Thus we don't care about the return value.
+            prompt_for_hazardous(e.target_name, e.cmd_name, e.hazardous_description)
+            target_name, cmd_name, cmd_params = $api_server.method_missing(cmd_no_hazardous, *args, timeout: timeout, log_message: log_message, validate: validate, scope: scope, token: token)
+            if log_message.nil? or log_message
```

Comment:
This is a weird check but I think it means log_message must be explicitly `false` for us to NOT call `_log_cmd`